File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/recommendation/modeling/dimension/model/ActivityParameterRecommendationDimension.ts
import {RecommendationDimension} from './RecommendationDimension';
import {AbstractActivityParameter, ParameterType} from '../../../../erc-data';

export class ActivityParameterRecommendationDimension extends RecommendationDimension {
  activityParameter: AbstractActivityParameter = new AbstractActivityParameter();
}

File: backend/recpro-backend-lib/src/main/java/de/ubt/ai4/petter/recpro/recpro_backend_lib/recpro/model/recommendation/modeling/dimension/ActivityParameterRecommendationDimension.java
package de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.recommendation.modeling.dimension;

import de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.ercData.activityParameter.modeling.AbstractActivityParameter;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
@Entity

@DiscriminatorValue("ACTIVITY_PARAMETER")
public class ActivityParameterRecommendationDimension extends RecommendationDimension {

    @ManyToOne
    private AbstractActivityParameter activityParameter;
}
